workflow/v1_2/workflow.go

```
@@ -78,21 +78,99 @@ type bakedInFile struct {
 	} `json:"bytes"`
 }
 
+// maybeMigrateFileParam returns either the original json representations, or a json representation as serialized files, if possible.
 func maybeMigrateFileParam(fm *effects.FileManager, param json.RawMessage) (json.RawMessage, error) {
-	bif := &bakedInFile{}
-	if err := json.Unmarshal([]byte(param), bif); err != nil {
-		return param, nil
-	} else if bif.Name == "" && len(bif.Bytes.Bytes) == 0 {
+
+	if success, js, _ := maybeMigrateFileFlat(fm, param); success {
+		return js, nil
+	} else if success, js, _ := maybeMigrateFileArray(fm, param); success {
+		return js, nil
+	} else if success, js, _ := maybeMigrateFileMap(fm, param); success {
```

Yeah, ignoring all the errors here does not seem the right thing to do at all as this will mask errors when writing out files. Combining success-bool with error would make this simpler.